tests/jax/test_distributed_layernorm_mlp.py

```diff
@@ -0,0 +1,266 @@
+# Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+import pytest
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+from typing import Callable, Sequence, Union
+
+from jax.sharding import Mesh, NamedSharding, PartitionSpec
+""" from distributed_test_base import compare_ops """
+from transformer_engine.jax.fp8 import FP8MetaPackage, FP8Helper
+from transformer_engine.jax.fp8 import is_fp8_available
+from transformer_engine.jax import fp8_autocast
+from transformer_engine.jax.flax import LayerNormMLP
+from transformer_engine.jax.mlp import fused_layernorm_fp8_mlp
+from transformer_engine.jax.sharding import HIDDEN_AXES, HIDDEN_TP_AXES, \
+    BATCH_AXES, SEQLEN_TP_AXES, SEQLEN_AXES, \
+    W_NO_SHARD_AXES, W_FSDP_AXES, W_TP_AXES, W_JOINED_AXES
+from transformer_engine.jax.sharding import MeshResource
+from utils import assert_allclose, is_devices_enough
+
+
+is_fp8_supported, reason = is_fp8_available()
+DTYPES = [jnp.bfloat16, jnp.float16]
+INPUT_SHAPE = [[64, 64, 32]]    # [seqlen, batch, hidden_in]
+
+LAYERNORM_INPUT_AXES = (BATCH_AXES, SEQLEN_TP_AXES, HIDDEN_AXES)
+DOT_1_INPUT_AXES = (BATCH_AXES, SEQLEN_AXES, HIDDEN_AXES)
+DOT_2_INPUT_AXES = (BATCH_AXES, SEQLEN_AXES, HIDDEN_TP_AXES)
+INTERMEDIATE = 16
+
+
+def generate_fsdp_and_tp_configs():
+    configs = []
+    if is_devices_enough(2):
+        configs.append(
+            [2, (1, 2), ('fsdp', 'tp'),
+             MeshResource(fsdp_resource='fsdp', tp_resource='tp')])
+
+    if is_devices_enough(4):
+        configs.append(
+            [4, (2, 2), ('fsdp', 'tp'),
+             MeshResource(fsdp_resource='fsdp', tp_resource='tp')])
+    return configs
+
+
+class TestDistributedLayernormMLP:
+
+    def generate_inputs(self, input_shape, activation_type, use_bias, dtype):
+        # m, n, k = shape
+        batch, seqlen, hidden_in = input_shape
```

Comment:
Here is not aligned to the comment of line 29. Here is b, s, h, but the comment of line 29 is s, b, h.